inlong-audit/audit-source/bin/flume-ng

```diff
@@ -0,0 +1,379 @@
+    IFS=$'\n'
+    for line in $HBASE_JAVA_LIBRARY_PATH; do
+      if [[ $line =~ ^java\.library\.path=(.*)$ ]]; then
+        HBASE_JAVA_LIBRARY_PATH=${BASH_REMATCH[1]}
+        break
+      fi
+    done
+    unset IFS
+
+    if [ -n "${HBASE_JAVA_LIBRARY_PATH}" ]; then
+      FLUME_JAVA_LIBRARY_PATH="$FLUME_JAVA_LIBRARY_PATH:$HBASE_JAVA_LIBRARY_PATH"
+    fi
+
+    # determine HBASE classpath
+    HBASE_CLASSPATH=$($HBASE_IN_PATH classpath)
+
+    # hack up and filter HBASE classpath
+    local ELEMENTS=$(sed -e 's/:/ /g' <<<${HBASE_CLASSPATH})
+    local ELEMENT
+    for ELEMENT in $ELEMENTS; do
+      local PIECE
+      for PIECE in $(echo $ELEMENT); do
+        if [[ $PIECE =~ slf4j-(api|log4j12).*\.jar ]]; then
+          info "Excluding $PIECE from classpath"
+          continue
+        else
+          FLUME_CLASSPATH="$FLUME_CLASSPATH:$PIECE"
+        fi
+      done
+    done
+    FLUME_CLASSPATH="$FLUME_CLASSPATH:$HBASE_HOME/conf"
+
+  fi
+}
+
+set_LD_LIBRARY_PATH(){
+#Append the FLUME_JAVA_LIBRARY_PATH to whatever the user may have specified in
+#flume-env.sh
+  if [ -n "${FLUME_JAVA_LIBRARY_PATH}" ]; then
+    export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${FLUME_JAVA_LIBRARY_PATH}"
+  fi
+}
+
+display_help() {
+  cat <<EOF
+Usage: $0 <command> [options]...
+
+commands:
+  help                  display this help text
+  agent                 run a Flume agent
+  avro-client           run an avro Flume client
+  version               show Flume version info
+
+global options:
+  --conf,-c <conf>      use configs in <conf> directory
+  --classpath,-C <cp>   append to the classpath
+  --dryrun,-d           do not actually start Flume, just print the command
+  -Dproperty=value      sets a JDK system property value
+
+agent options:
+  --conf-file,-f <file> specify a config file (required)
+  --name,-n <name>      the name of this agent (required)
+  --help,-h             display help text
+
+avro-client options:
+  --dirname <dir>       directory to stream to avro source
+  --host,-H <host>      hostname to which events will be sent (required)
+  --port,-p <port>      port of the avro source (required)
+  --filename,-F <file>  text file to stream to avro source [default: std input]
+  --headerFile,-R <file> headerFile containing headers as key/value pairs on each new line
+  --help,-h             display help text
+
+Note that if <conf> directory is specified, then it is always included first
+in the classpath.
+
+EOF
+}
+
+run_flume() {
+  local FLUME_APPLICATION_CLASS
+
+  if [ "$#" -gt 0 ]; then
+    FLUME_APPLICATION_CLASS=$1
+    shift
+  else
+    error "Must specify flume application class" 1
+  fi
+
+  if [ ${CLEAN_FLAG} -ne 0 ]; then
+    set -x
+  fi
+  $EXEC $JAVA_HOME/bin/java $JAVA_OPTS -cp "$FLUME_CLASSPATH" -Djava.util.logging.config.file="/data/work/osgoo/authcentertest/apache-flume-1.6.0-bin/conf/netty.properties" \
+      -Djava.library.path=$FLUME_JAVA_LIBRARY_PATH "$FLUME_APPLICATION_CLASS" $*
```

Comment:
the `java.util.logging.config.file` path may be invalid.

Reply:
ok